keystore/src/connection/platform/wasm/migration.rs

```diff
@@ -0,0 +1,198 @@
+use crate::connection::platform::wasm::version_number;
+use crate::connection::storage::{WasmEncryptedStorage, WasmStorageWrapper};
+use crate::connection::KeystoreDatabaseConnection;
+use crate::entities::{
+    E2eiAcmeCA, E2eiCrl, E2eiEnrollment, E2eiIntermediateCert, E2eiRefreshToken, Entity, EntityBase, MlsCredential,
+    MlsEncryptionKeyPair, MlsEpochEncryptionKeyPair, MlsHpkePrivateKey, MlsKeyPackage, MlsPendingMessage, MlsPskBundle,
+    MlsSignatureKeyPair, PersistedMlsGroup, PersistedMlsPendingGroup, ProteusIdentity, ProteusPrekey, ProteusSession,
+};
+use crate::{CryptoKeystoreError, CryptoKeystoreResult};
+use idb::builder::{DatabaseBuilder, IndexBuilder, ObjectStoreBuilder};
+use idb::KeyPath;
+use keystore_v_1_0_0::connection::KeystoreDatabaseConnection as KeystoreDatabaseConnectionV1_0_0;
+use keystore_v_1_0_0::entities::{
+    Entity as EntityV1_0_0, EntityFindParams as EntityFindParamsV1_0_0, MlsCredential as MlsCredentialV1_0_0,
+};
+use keystore_v_1_0_0::Connection as ConnectionV1_0_0;
+
+/// This is called from a while loop. The `from` argument represents the version the migration is performed from.
+/// The function will return the version number of the DB resulting from the migration.
+///
+/// To add a new migration, adjust the previous bottom match arm to return the current version,
+///     add a new match arm below that matches on that version, perform the migration workload
+///     and finally return `final_target`.
+pub(crate) async fn migrate(from: u32, final_target: u32, name: &str, key: &str) -> CryptoKeystoreResult<u32> {
+    const VERSION_NUMBER_V1_0_2: u32 = version_number(1, 0, 2, 0);
```

Comment:
This should be pushed down to the particular migration function (see below comment).